gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java

```diff
@@ -50,34 +50,43 @@
   private final String datePattern;
   private final Period lookbackPeriod;
   private final boolean isPatternHourly;
+  private final boolean isPatternMinutely;
   private final LocalDateTime currentTime;
 
   public TimeAwareRecursiveCopyableDataset(FileSystem fs, Path rootPath, Properties properties, Path glob) {
     super(fs, rootPath, properties, glob);
     this.lookbackTime = properties.getProperty(LOOKBACK_TIME_KEY);
-    PeriodFormatter periodFormatter = new PeriodFormatterBuilder().appendDays().appendSuffix("d").appendHours().appendSuffix("h").toFormatter();
+    PeriodFormatter periodFormatter = new PeriodFormatterBuilder().appendDays().appendSuffix("d").appendHours().appendSuffix("h").appendMinutes().appendSuffix("m").toFormatter();
     this.lookbackPeriod = periodFormatter.parsePeriod(lookbackTime);
     this.datePattern = properties.getProperty(DATE_PATTERN_KEY);
-    this.isPatternHourly = isDatePatternHourly(datePattern);
+    this.isPatternMinutely = isDatePatternMinutely(datePattern);
+    this.isPatternHourly = !this.isPatternMinutely && isDatePatternHourly(datePattern);
     this.currentTime = properties.containsKey(DATE_PATTERN_TIMEZONE_KEY) ? LocalDateTime.now(
         DateTimeZone.forID(DATE_PATTERN_TIMEZONE_KEY))
         : LocalDateTime.now(DateTimeZone.forID(DEFAULT_DATE_PATTERN_TIMEZONE));
 
-    //Daily directories cannot have a "hourly" lookback pattern. But hourly directories can accept lookback pattern with days.
-    if (!this.isPatternHourly) {
-      Assert.assertTrue(isLookbackTimeStringDaily(this.lookbackTime), "Expected day format for lookback time; found hourly format");
+    if (!this.isPatternMinutely) {
+      //Daily directories cannot have a "hourly" lookback pattern. But hourly directories can accept lookback pattern with days.
+      if (!this.isPatternHourly) {
+        Assert.assertTrue(isLookbackTimeStringDaily(this.lookbackTime), "Expected day format for lookback time; found hourly or minutely format");
+      } else {
+        //hourly directories cannot have a "minutely" lookback pattern. But minutely directories can accept lookback pattern with hours.
+        Assert.assertTrue(isLookbackTimeStringHourly(this.lookbackTime), "Expected hourly format for lookback time; found minutely format");
+      }
     }
   }
 
   public static class DateRangeIterator implements Iterator {
     private LocalDateTime startDate;
     private LocalDateTime endDate;
     private boolean isDatePatternHourly;
+    private boolean isDatePatternMinutely;
 
-    public DateRangeIterator(LocalDateTime startDate, LocalDateTime endDate, boolean isDatePatternHourly) {
+    public DateRangeIterator(LocalDateTime startDate, LocalDateTime endDate, boolean isDatePatternHourly, boolean isDatePatternMinutely) {
```

Comment:
Can we change the signature of this method to pass an Enum instead of 2 booleans?